nautobot/docs/plugins/region-and-site-data-migration-guide.md

```diff
@@ -0,0 +1,246 @@
+    for example_model in example_models:
+        site_name = example_model.site.name
+        # Point the location field to the corresponding "Site" LocationType Location
+        # with the same name.
+        example_model.location = Location.objects.get(location_type=site_location_type, name=site_name)
+    ExampleModel.objects.bulk_update(example_models, ["location"], 1000)
+```
+
+Finally, we need to add `migrations.RunPython` to the `operations` attribute in the migration class to execute this function when the migration is applied:
+
+```python
+    operations = [
+        migrations.RunPython(
+            # Execute the function
+            code=migrate_example_model_data_to_locations,
+            reverse_code=migrations.operations.special.RunPython.noop,
+        )
+    ]
+
+```
+
+The final migration file might look like this:
+
+```python
+# Generated by Django 3.2.17 on 2023-02-22 15:38
+# 0008_migrate_example_model_data_from_site_to_location
+
+from django.db import migrations
+
+def migrate_example_model_data_to_locations(apps, schema_editor):
+
+    ExampleModel = apps.get("example_app", "examplemodel")
+    LocationType = apps.get("dcim", "locationtype")
+    Location = apps.get("dcim", "location")
+    # Get "Site" LocationType
+    site_location_type = LocationType.objects.get(name="Site")
+
+    if  ExampleModel.objects.exists():
+        # Query ExampleModel instances with non-null site field
+        example_models = ExampleModel.objects.filter(site__isnull=False).select_related("site", "location")
+        for  example_model  in  example_models:
+            site_name = example_model.site.name
+            # Point the location field to the corresponding "Site" LocationType Location
+            # with the same name.
+            example_model.location = Location.objects.get(location_type=site_location_type, name=site_name)
+        ExampleModel.objects.bulk_update(example_models, ["location"], 1000)
+
+class Migration(migrations.Migration):
+    dependencies = [
+        # The dcim migration creates the Site Type and Region Type Locations that
+        # your data models are migrating to. It has to be run **before** this migration.
+        ("dcim", "0030_migrate_region_and_site_data_to_locations"),
+        ("example_app", "0007_add_location_field_to_example_model"),
+    ]
+    operations = [
+        migrations.RunPython(
+            # Execute the function
+            code=migrate_example_model_data_to_locations,
+            reverse_code=migrations.operations.special.RunPython.noop,
+        )
+    ]
+```
+
+## Remove Site/Region Related Fields from Migrated Data Models
+
+After the data migration is successful, we need to remove the `site`/`region` fields from your data model so that Nautobot will be able to remove the `Site` and `Region` models. You can do that by simply removing the `site`/`region` attributes from your model class:
```

Comment:
Maybe call out here that this needs to go in a separate migration file from the first, since the first is migrating data and this is changing the schema?